npc/kafras/functions_kafras.txt

```diff
@@ -267,16 +267,51 @@ function	script	F_KafStor	{
 		}
 		// Normal message
 		else {
-			mes "[Kafra Employee]";
-			mes "Here, let me open";
-			mes "your Storage for you.";
-			mes "Thank you for using";
-			mes "the Kafra Service.";
+			if (!checkre(RENEWAL)) {
+				mes "[Kafra Employee]";
+				mes "Here, let me open";
+				mes "your Storage for you.";
+				mes "Thank you for using";
+				mes "the Kafra Service.";
+			} else {
+				mes "[Kafra Employee]";
+				mes "Please select the Storage you would like to open";
+				next;
+				.@storage = select("Storage 1:Storage 2:Storage 3");
+				mes "[Kafra Employee]";
+				mes "Here, let me open";
+				mes "your Storage for you.";
+				mes "Thank you for using";
+				mes "the Kafra Service.";
+			}
 		}
 	}
 	callfunc("F_CheckKafCode");	//check your storage password, if set
 	close2;
-	openstorage;
+	if (!checkre(RENEWAL)) {
+		openstorage;
+	} else {
+		switch (.@storage) {
```

Comment:
This switch could be in a function, that way the storage from https://github.com/rathena/rathena/pull/8678/files#diff-5c3e6b392ff6e40c2589b1c5da66cf7518406033d56c8cb240414e338197e837R124
could be called using
`callfunc("F_openstorage",1);` (open the storage 1)